packages/find-root/src/index.ts

```diff
@@ -80,60 +60,44 @@ export async function findRoot(cwd: string): Promise<MonorepoRoot> {
         .then((result) => {
           if (result) {
             monorepoRoot = result;
-            return dir;
+            return directory;
```

Comment:
This change is interesting - this is likely a bug fix for which we don't have a test case. It was always returning `undefined` from here since `dir` was declared after `findUp` call. 

Most likely this fixes an issue with nested monorepos. Before this fix, it's likely that the outer one would be found whereas with this change we correctly find the nearest one.